web/src/types/index.ts

```diff
@@ -60,43 +49,79 @@ type AppStatus = {
   updatedAt: string;
 };
 
+export type Credentials = {
+  username: string;
+  password: string;
+};
+
+export type DashboardActionLink = {
+  title: string;
+  uri: string;
+};
+
 export type DashboardResponse = {
   appStatus: AppStatus | null;
   metrics: string[];
   prometheusAddress: string;
 };
 
 export type Downstream = {
+  cluster: string;
   currentVersion: Version;
   gitops: GitOps;
   links: DashboardActionLink[];
   pendingVersions: Version[];
 };
 
+export type Entitlement = {
+  title: string;
+  value: string;
+  label: string;
+  valueType: "Text" | "Boolean" | "Integer" | "String";
+};
+
 export type GitOps = {
   isConnected: true;
   uri: string;
 };
 
 export type KotsParams = {
+  owner: string;
   sequence: string;
   slug: string;
+  tab: string;
 };
 
-export type DashboardActionLink = {
-  title: string;
-  uri: string;
+export type Metadata = {
+  isAirgap: boolean;
+  isKurl: boolean;
+};
+
+export type ResourceStates = {
+  kind: string;
+  name: string;
+  namespace: string;
+  // from https://github.com/replicatedhq/kots/blob/84b7e4e0e9275bb200a36be69691c4944eb8cf8f/pkg/appstate/types/types.go#L10-L14
+  state: "ready" | "updating" | "degrading" | "unavailable" | "missing";
+};
+
+export type ThemeState = {
+  navbarLogo: string | null;
 };
 
 export type Version = {
   parentSequence: number;
   semver: string;
   sequence: number;
+  status: VersionStatus;
   versionLabel?: string;
 };
-export type Entitlement = {
-  title: string;
-  value: string;
-  label: string;
-  valueType: "Text" | "Boolean" | "Integer" | "String";
-};
+
+type VersionStatus =
+  | "deploying"
+  | "failed"
+  | "pending_config"
+  | "pending_download"
+  | "pending_preflight"
+  | "pending_app"
+  | "deployed";
```

Comment:
new types: 
- VersionStatus
- ThemeState
- Metadata

The remainder of the changes are alphabetizing the types file

Reply:
I'll make sure to keep things alphabetized.